File: src/main/java/cn/labzen/web/ap/processor/CreativeProcessor.java
package cn.labzen.web.ap.processor;

import cn.labzen.web.ap.internal.ClassCreator;
import cn.labzen.web.ap.internal.context.ControllerContext;

public final class CreativeProcessor implements InternalProcessor {

  @Override
  public void process(ControllerContext context) {
    new ClassCreator(context.getRoot(), context.getApc().filer()).create();
  }

  @Override
  public int priority() {
    return PRIORITY_CREATIVE;
  }
}

File: src/main/kotlin/cn/labzen/web/ap/processor/CreativeProcessor.kt
package cn.labzen.web.ap.processor

import cn.labzen.web.ap.internal.ClassCreator
import cn.labzen.web.ap.internal.context.ControllerContext
import cn.labzen.web.ap.processor.InternalProcessor.Companion.PRIORITY_CREATIVE

class CreativeProcessor : InternalProcessor {

  override fun process(context: ControllerContext) {
    ClassCreator(context.root, context.apc.filer).create()
  }

  override fun priority(): Int = PRIORITY_CREATIVE
}